clients/drcachesim/tracer/instru.cpp

```diff
@@ -181,17 +181,67 @@ instru_t::instr_to_prefetch_type(instr_t *instr)
     }
 }
 
+#ifdef AARCH64
+bool
+instru_t::is_aarch64_icache_flush_op(instr_t *instr)
+{
+    if (instr_get_opcode(instr) != OP_sys)
+        return false;
+    switch (opnd_get_immed_int(instr_get_src(instr, 0))) {
+    case DR_IC_IVAU: return true;
+    }
+    return false;
+}
+
+bool
+instru_t::is_aarch64_dcache_flush_op(instr_t *instr)
+{
+    if (instr_get_opcode(instr) != OP_sys)
+        return false;
+    switch (opnd_get_immed_int(instr_get_src(instr, 0))) {
+    case DR_DC_CVAU:
+    case DR_DC_CIVAC:
+    case DR_DC_CVAC: return true;
```

Comment:
Unlike other DC operations handled here, `DC ZVA` does not invalidate or clear the cache line. IIUC, it's not a cache flush operation.

 It just zeroes out a region of memory. So maybe it can be treated as a store?

Reply:
It does seem like it could be treated as a regular store.  But it needs a proper start address (back-align its operand) and size (cache line size).  The easiest thing for a client would be to have that in the IR operand, instead of the encoded address: which bring up issues of re-encoding fidelity (xref #3339).  Grrr, another source of complexity.  I filed #4400.

On a similar topic: I'm thinking the size of the cache operations (here, and OP_clflush) should match OP_invlpg and have a 0 size, and that we should change opc_is_not_a_real_memory_load() to just look for zero-sized memory operand?  I think most tools (taint tracking is always the example I think of) would not want to think of these as reading a byte of memory: they should not interact w/ the taint of data I would think.  I filed #4401.

Reply:
As this PR is for cache flush instructions, I'll create a new issue and PR for handling `DC ZVA`.

Reply:
I can continue on `DC ZVA` work under #4400 itself.